File: matriz/exercicio5.py
alunos = [[0]*4 for _ in range(5)]

# (a) Ler matrícula, média das provas e média dos trabalhos
for i in range(5):
    print(f"Aluno {i+1}:")
    alunos[i][0] = int(input("Número da matrícula: "))
    alunos[i][1] = int(input("Média das provas: "))
    alunos[i][2] = int(input("Média dos trabalhos: "))

    # (b) Calcular nota final (soma das médias)
    alunos[i][3] = alunos[i][1] + alunos[i][2]

# (c) Encontrar matrícula do aluno com maior nota final
maior_nota = alunos[0][3]
matricula_maior = alunos[0][0]

for i in range(1, 5):
    if alunos[i][3] > maior_nota:
        maior_nota = alunos[i][3]
        matricula_maior = alunos[i][0]

# (d) Calcular média aritmética das notas finais
soma_notas = sum(alunos[i][3] for i in range(5))
media_notas = soma_notas / 5

# Resultados
print(f"\nMatrícula do aluno com maior nota final: {matricula_maior}")
print(f"Média aritmética das notas finais: {media_notas:.2f}")

File: matriz/exercicio5.c
#include <stdio.h>

int main() {
    int alunos[5][4]; 
    int i;
    int matricula_maior_nota = 0;
    int maior_nota = -1;
    float soma_notas = 0;

    // (a) 
    for (i = 0; i < 5; i++) {
        printf("Aluno %d\n", i+1);
        printf("Matricula: ");
        scanf("%d", &alunos[i][0]);
        printf("Media das provas: ");
        scanf("%d", &alunos[i][1]);
        printf("Media dos trabalhos: ");
        scanf("%d", &alunos[i][2]);

        // (b) 
        alunos[i][3] = alunos[i][1] + alunos[i][2];

        soma_notas += alunos[i][3];

        if (alunos[i][3] > maior_nota) {
            maior_nota = alunos[i][3];
            matricula_maior_nota = alunos[i][0];
        }
    }

    // (c) 
    printf("\nAluno com maior nota final: %d\n", matricula_maior_nota);

    // (d) 
    printf("Media aritmetica das notas finais: %.2f\n", soma_notas / 5);

    return 0;
}
